### PR TITLE
CI: fix set output warns and add caching

### PR DESCRIPTION
**Pull request by illusion0001**
~~Downgraded to LLVM 10 because LLVM 12 crashes when applying save patches~~ not needed
- Change multiline commands to single line
- Remove unnecessary env vars
- Add caches for big library (SDL and libcurl)

### CODE REVIEW

**Comment by bucanero** [comment]
About llvm12, have you tried with my latest commits? I'm using clang 12 locally and I have no issues.

I specifically added this change here that fixed the crashes with llvm-12 build on my local environment:
https://github.com/bucanero/apollo-ps4/blob/2d615b1fbf32de918b07f13ceb248c55b43d099f/source/orbis_jbc.c#L309

Now I'm copying the patch data to an aligned memory buffer first, then I call the goldhen syscall. I suspect that if memory is not properly aligned then writing to the process triggers a kernel panic.

Right now I'm using 16 bytes alignment, but if you still have issues I'd try 128 bytes or 256 to confirm if it works or not.

**Comment by illusion0001** [comment]
Will try it, specifically it crashed during shellcore patches on LLVM 12 using the CI builds.

**Comment by bucanero** [comment]
> Will try it, specifically the app crashed during shellcore patches on LLVM 12 using the CI builds.

let me know when you test, as I'd like to keep LLVM 12, since it's the recommended version by Open Orbis and the same version I'm using locally.
I can also share my own local build with clang 12 on macos (that is not crashing) in case you want to compare.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -65,19 +72,22 @@ jobs:
...
-      # temporary release until 0.53 is released
+    # temporary release until 0.53 is released
     - name: Download OpenOrbis Toolchain
+      if: steps.cache-oosdk.outputs.cache-hit != 'true'
       run: |
-        curl -sL https://github.com/illusion0001/OpenOrbis-PS4-Toolchain/releases/download/0.0.1.416/toolchain.tar.gz | tar xvz -C ./
-        echo "OO_PS4_TOOLCHAIN=${GITHUB_WORKSPACE}/OpenOrbis/PS4Toolchain" >> $GITHUB_ENV
-        cp oosdk_libraries/build_rules.mk OpenOrbis/PS4Toolchain/build_rules.mk
+        curl -fLOs https://github.com/illusion0001/OpenOrbis-PS4-Toolchain/releases/download/0.0.1.416/toolchain.tar.gz
+        tar xf toolchain.tar.gz
```
Comment (bucanero) [inline .github/workflows/build.yml:87]:
```suggestion
        curl -sL https://github.com/illusion0001/OpenOrbis-PS4-Toolchain/releases/download/0.0.1.416/toolchain.tar.gz | tar xvz -C ./
```

We can decompress on the fly, no need to keep the .tar.gz file on disk.

**Comment by illusion0001** [comment]
@bucanero tried it.
```c
 void* data = aligned_alloc(64, 256);
 ```
 64 bytes seems fine, if I do 16 bytes app crashes.

**Comment by illusion0001** [comment]
Nevermind, it stopped crashing on LLVM 12 as well

**Comment by bucanero** [comment]
> Nevermind, it stopped crashing on LLVM 12 as well

so both 0x10 and 0x40 worked with llvm-12? I guess memory must be aligned for these goldhen/ps4debug syscalls after all. 

I'll try 64 on my side but if 0x10 works then we'll keep that.

**Comment by illusion0001** [comment]
Yea, no idea what's wrong. grabbed newest CI build and app was crashing when applying save patches now it does not. maybe because I had ps4debug payload running

**Comment by illusion0001** [comment]
Nope, ps4debug loaded and relaunched the app, llvm12 doesn't crash anymore. PS4 is a strange device

**Comment by illusion0001** [comment]
Redownloaded the build that was crashing before and it now doesn't crash too. LLVM change is not needed,
https://github.com/bucanero/apollo-ps4/actions/runs/3560591420

**Comment by illusion0001** [comment]
> so both 0x10 and 0x40 worked with llvm-12?

yes

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -94,75 +111,85 @@ jobs:
...
+      working-directory: ${{ env.curl_path }}
       run: |
         mkdir orbis && cd orbis
         cmake --toolchain ../../../SDL-PS4/cmake/openorbis.cmake .. -DCMAKE_USE_POLARSSL=1 -DUSE_UNIX_SOCKETS=0 -DENABLE_THREADED_RESOLVER=0 -DENABLE_IPV6=0
         make libcurl
-        cp lib/libcurl.a "${OO_PS4_TOOLCHAIN}/lib"
-        cp -R ../include/curl "${OO_PS4_TOOLCHAIN}/include/"
 
-    - name: Build Apollo App Package
+    - name: Copy lib
```
Comment (bucanero) [inline .github/workflows/build.yml:178]:
```suggestion
    - name: Copy cached libraries
```

**Review by bucanero — APPROVED** [review]

**Review by bucanero — APPROVED** [review]

**Pull request merged**